L1Trigger/L1TTrackMatch/plugins/L1TkHTMissEmulatorProducer.cc

```diff
@@ -0,0 +1,286 @@
+      }
+      else{
+	tmp_jet_sin_phi = -1*cosLUT_[-1 * (cosLUTbins/2 + 1 - (-1*tmp_jet_phi))];
+	}
+    }
+
+    l1tmhtemu::Et_t tmp_jet_px = tmp_jet_pt*tmp_jet_cos_phi;
+    l1tmhtemu::Et_t tmp_jet_py = tmp_jet_pt*tmp_jet_sin_phi;
+
+    jetn++;
+
+
+    if(debug_){
+      edm::LogVerbatim("L1TrackerHTMissEmulatorProducer")
+      << "****JET EMULATION"<< jetn << "****\n"
+      << "FLOATS ORIGINAL\n" << "PT: " << jetIter->pt() << "| ETA: " << jetIter->glbeta() << "| PHI: " << jetIter->glbphi() << "| NTRACKS: " << jetIter->nt()<< "| COS(PHI): " << cos(jetIter->glbphi()) << "| SIN(PHI): "<< sin(jetIter->glbphi())<< "| Px: " << jetIter->pt()*cos(jetIter->glbphi()) << "| Py: " << jetIter->pt()*sin(jetIter->glbphi())<< "\n"
+      <<"AP_INTS RAW\n" << "PT: " << jetIter->ptWord() << "| ETA: " << jetIter->glbEtaWord() << "| PHI: " << jetIter->glbPhiWord() << "| NTRACKS: " << jetIter->ntWord()<< "\n"
+      <<"AP_INTS NEW\n"<< "PT: " << tmp_jet_pt << "| ETA: " << tmp_jet_eta << "| PHI: " << tmp_jet_phi << "| NTRACKS: " << tmp_jet_nt << "| COS(PHI): "<< tmp_jet_cos_phi << "| SIN(PHI): " << tmp_jet_sin_phi<< "| Px: " << tmp_jet_px << "| Py: " << tmp_jet_py << "\n"
+      <<"AP_INTS NEW TO FLOATS\n"<< "PT: " << (float) tmp_jet_pt * l1tmhtemu::kStepPt << "| ETA: " << (float) tmp_jet_eta * l1tmhtemu::kStepEta << "| PHI: " << (float) tmp_jet_phi * l1tmhtemu::kStepPhi << "| NTRACKS: " << (int) tmp_jet_nt << "| COS(PHI): "<< (float) tmp_jet_cos_phi * l1tmhtemu::kStepPhi << "| SIN(PHI): " << (float) tmp_jet_sin_phi * l1tmhtemu::kStepPhi << "| Px: " << (float) tmp_jet_px * l1tmhtemu::kStepPt * l1tmhtemu::kStepPhi << "| Py: " << (float) tmp_jet_py * l1tmhtemu::kStepPt * l1tmhtemu::kStepPhi << "\n"
+    << "-------------------------------------------------------------------------\n";
+    }
+
+
+    if(debug_){
+      sumPxtemp_ += tmp_jet_px_;
+      sumPytemp_ += tmp_jet_py_;
+      HTtemp_ += tmp_jet_pt_;
+    }
+
+    if (tmp_jet_pt < jetMinPt_)
+      continue;
+    if (tmp_jet_eta > jetMaxEta_ or tmp_jet_eta < -1*jetMaxEta_)
+      continue;
+    if (tmp_jet_nt < minNtracksLowPt_ && tmp_jet_pt > 200)
+      continue;
+    if (tmp_jet_nt < minNtracksHighPt_ && tmp_jet_pt > 400)
+      continue;
+
+    jetnpasscuts++;
+
+    if(debug_){
+      sumPx_ += tmp_jet_px_;
+      sumPy_ += tmp_jet_py_;
+      HT_ += tmp_jet_pt_;
+    }
+
+    sumPx += tmp_jet_pt * tmp_jet_cos_phi;
+    sumPy += tmp_jet_pt * tmp_jet_sin_phi;
+    HT += tmp_jet_pt;
+
+  }  // end jet loop
+
+  // define missing HT
+  // Perform cordic sqrt, take x,y and converts to polar coordinate r,phi where
+  // r=sqrt(x**2+y**2) and phi = atan(y/x)
+  int aSteps = 8;
+  l1tmhtemu::EtMiss EtMiss = l1tmhtemu::cordicSqrt(sumPx, sumPy, aSteps);
+  math::XYZTLorentzVector missingEt(-sumPx, -sumPy, 0, EtMiss.Et);
+
+  l1tmhtemu::MHTphi_t phi = 0;
+
+  if ((sumPx < 0) && (sumPy < 0))
+    phi = EtMiss.Phi - l1tmhtemu::kMHTPhiBins / 2;
```

Comment:
There is a lot of inherent casting to integers in this line. In particular, the remainder from the right hand side of the equation is lost since `phi` is of integer type. Is this intentional?

Reply:
The right hand side `l1tmhtemu::kMHTPhiBins / 2` is actually an integer by construction 2^14 / 2. Hence, there is no remainder loss.